main.py

```diff
@@ -126,22 +143,48 @@ def on_click(self):
                     aircraft.label()
         if event.type == pygame.MOUSEBUTTONDOWN:
             x, y = pygame.mouse.get_pos()
-            print(x)
-            print(y)
+            collide = False
             # Detect aircraft clicks
             for aircraft in aircrafts:
                 if aircraft.rect.collidepoint(x, y):
-                    print("collide")
-                    aircraft.on_click()
+                    collide = True
+                    selected = aircraft
+                    # Update all aircraft colors
+                    for aircraft in aircrafts:
+                        aircraft.label()
 
+                    altitude_dropdown.selected_option = str(selected.altitude)
+                    altitude_dropdown.show()
```

Comment:
This is the issue I was talking about, when I change the selected option on 156 it doesnt actually change the selected option until you open the selector. Not sure if this is an issue with the package or if I am missing something stupid.

Reply:
I believe [selected_option](https://pygame-gui.readthedocs.io/en/latest/pygame_gui.elements.html) is working as intended/documented, in that the current altitude of the object is selected (seen when you click the dropdown). It's the initial text (which doesn't seem to be something we can edit?) that needs to be changed... similar to how if you select the "6000" altitude plane first, the drop_down_menu stays blank.

I considered destroying/rebuilding the drop_down_menu instead of hiding, but that would just cause the seen value to be blank when you first click (correcting once you click the arrow).

I think we may need to look into an alternative method (not sure what right now)...

Reply:
Ahh, that makes some sense. Still doesn't explain why when you click on another aircraft it doesn't update, or maybe it does!

We could potentially use the UIClosedDropDownState, it might have what we need? I dont really understand it, the docs aren't particularly clear.

Reply:
I think it's because we're using the same menu for all aircrafts, right? I think we need to create a separate menu per aircraft... perhaps a separate class that creates the menu or a method inside the aircraft class? Does that make sense?

Reply:
Yeah that makes sense, I was planning on doing this really anyway, I was just putting it off haha. I will work on this this morning.